### PR TITLE
[SPARKC-524] Generate a CQL Create Table from DataFrame with options

### PR DESCRIPTION
**Pull request by alexott**
Generate a CQL Create Table statement using a DataFrame and all the parameters required to
construct the CREATE Table statement, like, `IF NOT EXISTS`, `WITH CLUSTERING ORDER BY`,
etc.

### CODE REVIEW

**Comment by RussellSpitzer** [comment]
This looks good to me, we had a discussion of a few things though to expand it on slack:

* Maybe use the ClusteringOrdering class for ASC Desc? (not Boolean)
* Possible to add TableOptions Map<String, String> for generic other options

**Comment by alexott** [comment]
Changes are made as discussed, plus added a function for generation of table from DataFrame - previous version was only for RDDs...

File: connector/src/main/scala/com/datastax/spark/connector/DatasetFunctions.scala [hunk 1]
```diff
@@ -54,18 +76,20 @@ class DatasetFunctions[K: Encoder](dataset: Dataset[K]) extends Serializable {
...
           .zipWithIndex
-          .map { case (col, index) => col.copy(columnRole = ClusteringColumn(index))}
+          .map { case (col, index) => col._1.copy(columnRole = ClusteringColumn(index, col._2))}
       ,
       regularColumns = regularColumnNames
           .map(regularColumnName =>
              columnMapping.getOrElse(regularColumnName,
                   throw missingColumnException(regularColumnName, "Regular Column")))
           .map(_.copy(columnRole = RegularColumn))
+      , ifNotExists = ifNotExists, tableOptions = tableOptions
```
Comment (RussellSpitzer) [inline connector/src/main/scala/com/datastax/spark/connector/DatasetFunctions.scala]: usually we would have a carriage return between args

Comment (alexott) [reply connector/src/main/scala/com/datastax/spark/connector/DatasetFunctions.scala]: let me push the change with CR between args...

**Comment by RussellSpitzer** [comment]
Thanks! I was going to just do it for you and merge but this makes my life
easier :P

**Comment by RussellSpitzer** [comment]
Thanks @alexott

**Pull request merged**